### ISSUE TITLE
Sharing a non-shareable file yields 'could not create share'

### ISSUE THREAD

**Issue by lnxbil** [Zimbra-Community/owncloud-zimlet#169]
Hi Berry,

If I want to share a file that is prohibited from sharing by the original author/sharer, then the Zimlet will display 'Could not create share'. Unfortunately, I cannot see any reason in the returned response besides the error message itself:

```json
{
	"Header": {
		"context": {
			"session": {
				"id": "1184",
				"_content": "1184"
			},
			"change": {
				"token": 5824
			},
			"_jsns": "urn:zimbra"
		}
	},
	"Body": {
		"response": {
			"createShare": "{\"statuscode\":100,\"id\":0,\"message\":\"\",\"url\":\"Could not create share. \",\"status\":\"ok\",\"token\":\"\"}"
		}
	},
	"_jsns": "urn:zimbraSoap"
}
```

I do not know how to approach this in any kind, so I'm going to open an issue.

**Comment by barrydegraaff** [comment]
I understand the message is not very helpful for the user, however I think the message is pretty much all returned by the OCS api, I am not sure how to improve that, other than filing a bug over at nextcloud asking them to make the message better.

https://docs.nextcloud.com/server/13/developer_manual/core/ocs-share-api.html

Its probably a fix on the Nextcloud side...

**Comment by lnxbil** [comment]
Yes, I thought so too. Maybe also a bug on their side ... `statuscode: 100` signals success.

I'm going to open an issue on the nextcloud side.

**Comment by barrydegraaff** [comment]
Yeah but please note that the statuscode in the SOAP message is hardcoded to 100.

https://github.com/Zimbra-Community/OCS/blob/master/extension/src/tk/barrydegraaff/ocs/OCS.java#L200

The message is not, that is passed straight for NC to the user.

**Comment by lnxbil** [comment]
>Yeah but please note that the statuscode in the SOAP message is hardcoded to 100. 

Good to know ...

I'm just trying to recreate the share-call with `curl` to see what actually happens for a better bug report/enhancement for NC.

**Comment by barrydegraaff** [comment]
This should help you:
https://docs.nextcloud.com/server/13/developer_manual/client_apis/OCS/index.html

**Comment by lnxbil** [comment]
Hi Barry,

sorry it took me so long to answer, but I needed to setup a test environment for the plugin. I installed it and now I get this:

![image](https://user-images.githubusercontent.com/727711/39626399-ffa6b866-4fa1-11e8-8cb7-af42cd585086.png)

Yes, it's "more", but still not perfect. Thank you for your time.

> This should help you:
> https://docs.nextcloud.com/server/13/developer_manual/client_apis/OCS/index.html

Yes, I used that information to get it working . Thank you.

In the end, the API responds correctly and I do not know where the `illegalStateException` originates from:

```
curl -u $USERNAME:$PASSWORD -H "OCS-APIRequest: true" -X POST https://$HOST/ocs/v1.php/apps/files_sharing/api/v1/shares -d path=/Administration/Test.md -d shareType=3
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>404</statuscode>
  <message>You are not allowed to share /$USERNAME/files/Administration/Test.md</message>
  <totalitems></totalitems>
  <itemsperpage></itemsperpage>
 </meta>
 <data/>
</ocs>
```

**Comment by barrydegraaff** [comment]
Its a bug, not a very fatal one, so some day I will get around to it.
